src/terminal/index.ts

```diff
@@ -360,54 +353,59 @@ class TerminalWidget extends Widget {
    * Use the dummy terminal to measure the row and column sizes.
    */
   private _snapTermSizing(): void {
-    if (!this.isAttached || !this.isVisible) {
+    if (!this.isVisible) {
       this._dirty = true;
       return;
     }
-    let style = window.getComputedStyle(this.node);
-    this._widthOffset = parseInt(style.paddingLeft) + parseInt(style.paddingRight);
-    this._heightOffset = parseInt(style.paddingTop) + parseInt(style.paddingBottom);
     let node = this._dummyTerm;
     this._term.element.appendChild(node);
-    this._rowHeight = node.offsetHeight / DUMMY_ROWS;
-    this._colWidth = node.offsetWidth / DUMMY_COLS;
+    let offsetWidth = node.offsetWidth;
+    let offsetHeight = node.offsetHeight
+    this._rowHeight = offsetHeight / DUMMY_ROWS;
+    this._colWidth = offsetWidth / DUMMY_COLS;
     this._term.element.removeChild(node);
-    this._dirty = false;
-    if (this._width !== -1) {
-      this._resizeTerminal();
-    }
+    this._resizeTerminal(offsetWidth, offsetHeight);
   }
 
   /**
    * Resize the terminal based on the computed geometry.
+   *
+   * The parent offset dimensions should be `-1` if unknown.
    */
-  private _resizeTerminal() {
-    if (this._term === null) {
+  private _resizeTerminal(offsetWidth: number, offsetHeight: number) {
+    if (this._rowHeight === -1 || !this.isVisible) {
+      this._dirty = true;
       return;
     }
-    let height = this._height - this._heightOffset;
-    let width = this._width - this._widthOffset;
+    // Measure the parent if the offset dimensions are unknown.
```

Comment:
But you could also just ditch the comment. It's not needed when nothing else has comments.